File: Parameters/Standard/Components/ListParameterSettings.vb
Imports System.Xml.Serialization
Imports System.Web.UI.WebControls

Namespace DNNStuff.SQLViewPro.StandardParameters

    <XmlRootAttribute(ElementName:="Settings", IsNullable:=False)> _
    Public Class ListParameterSettings
        Public Property ConnectionId() As Integer
        Public Property Command() As String
        Public Property List() As String
        Public Property [Default]() As String
        Public Property CommandCacheTimeout() As Integer = 60
    End Class

    <XmlRootAttribute(ElementName:="Settings", IsNullable:=False)> _
    Public Class DropDownListParameterSettings
        Inherits ListParameterSettings
        Public Property AutoPostback() As Boolean = False
    End Class

    <XmlRootAttribute(ElementName:="Settings", IsNullable:=False)> _
    Public Class ListBoxParameterSettings
        Inherits ListParameterSettings
        Public Property AutoPostback() As Boolean = False
        Public Property MultiSelect() As Boolean = False
        Public Property MultiSelectSize() As Integer = 5
    End Class

    <XmlRootAttribute(ElementName:="Settings", IsNullable:=False)> _
    Public Class FlowListParameterSettings
        Inherits ListParameterSettings

        Public Property RepeatLayout() As RepeatLayout
        Public Property RepeatDirection() As RepeatDirection
        Public Property RepeatColumns() As Integer = 2
    End Class


End Namespace

File: Parameters/Standard/Components/ListParameterSettings.cs




using System.Xml.Serialization;
using System.Web.UI.WebControls;


namespace DNNStuff.SQLViewPro.StandardParameters
{
	
	[XmlRootAttribute(ElementName = "Settings", IsNullable = false)]public class ListParameterSettings
	{
		public int ConnectionId {get; set;}
		public string Command {get; set;}
		public string List {get; set;}
		public string Default {get; set;}
	    public int CommandCacheTimeout { get; set; } = 60;
	}
	
	[XmlRootAttribute(ElementName = "Settings", IsNullable = false)]public class DropDownListParameterSettings : ListParameterSettings
	{
	    public bool AutoPostback { get; set; } = false;
	}
	
	[XmlRootAttribute(ElementName = "Settings", IsNullable = false)]public class ListBoxParameterSettings : ListParameterSettings
	{
	    public bool AutoPostback { get; set; } = false;

	    public bool MultiSelect { get; set; } = false;

	    public int MultiSelectSize { get; set; } = 5;
	}
	
	[XmlRootAttribute(ElementName = "Settings", IsNullable = false)]public class FlowListParameterSettings : ListParameterSettings
	{
		
		public RepeatLayout RepeatLayout {get; set;}
		public RepeatDirection RepeatDirection {get; set;}
	    public int RepeatColumns { get; set; } = 2;
	}
	
	
}

